src/lib/crypto/snarky_tests/snarky_tests.ml

```diff
@@ -325,21 +328,30 @@ module As_prover_circuits = struct
   let main as_prover vars
       ((b1, b2, b3) : Impl.Field.t * Impl.Field.t * Impl.Field.t) () =
     let abc = Impl.Field.(b1 + b2 + b3) in
-    assert (get_id b1 = 1) ;
+
+    (* we encode the assumption that variables are indexed starting at zero
+       (which used not to be the case due to an extra R1CS row)
+    *)
+    assert (get_id b1 = 0) ;
+
+    (* if [as_prover] is set, we try to access all variables that have been created by index *)
     if as_prover then
       Impl.as_prover (fun _ ->
+          (* sum all variables *)
           let f acc e =
             let var = Snarky_backendless.Cvar.Unsafe.of_index e in
             let v = Impl.As_prover.read_var var in
             Impl.Field.Constant.(acc + v)
           in
-          let l = List.range 1 (vars + 1) in
+          let l = List.range 0 vars in
           let total : Impl.field =
-            List.fold l ~init:Impl.Field.Constant.one ~f
+            List.fold l ~init:Impl.Field.Constant.zero ~f
           in
-          assert (not (Impl.Field.(Constant.compare total Constant.one) = 0)) ;
-          assert (not Impl.Field.Constant.(equal total one)) ;
-          () ) ;
+
+          (* manual sum is equal to circuit sum *)
+          let abc = Impl.As_prover.read_var abc in
+
+          assert (Impl.Field.(Constant.equal abc total)) ) ;
```

Comment:
@fabrizio-m I slightly changed this test to check that I can read the sum `abc` and check that it's equal to the indexed variables. Now that I'm looking into this test I'm wondering why we're really testing these, as nobody should access variables manually like this (via `Unsafe.of_index`). Maybe we can delete this test in the future?

Reply:
I think the idea was to check that the indexing was consistent through both implementations.

Reply:
yeah that makes sense : o I've kept that behavior so it'll still get tested!